CI: BPF Checks --TestBPF/bpf_nat_tests.o/nat4_port_allocation_tcp

## CI failure

In the past week, ~2.5% of BPF Checks workflow runs have been failing with this error. 

```
--- FAIL: TestBPF (39.98s)
--- FAIL: TestBPF/bpf_nat_tests.o (1.43s)
--- FAIL: TestBPF/bpf_nat_tests.o/nat4_port_allocation_tcp (0.58s)

    bpf_test.go:480: bpf_nat_tests.c:1412: assert failed at bpf_nat_tests.c:1412

┌──────────────────────────────────────────────────────────────────────────────────────────────┐
│  STATUS │ ELAPSED │                  PACKAGE                   │ COVER │ PASS │ FAIL │ SKIP  │
│─────────┼─────────┼────────────────────────────────────────────┼───────┼──────┼──────┼───────│
│  FAIL   │ 40.01s  │ github.com/cilium/cilium/bpf/tests/bpftest │  --   │ 536  │  3   │  0    │
└──────────────────────────────────────────────────────────────────────────────────────────────┘
```

Looking at the code, this is the failing assertion:
```c
/* Only occasional failures at 50% of the test. */
assert(retries_50percent[SNAT_COLLISION_RETRIES] < 15);
```
This tests the port allocation algorithm in SNAT, which has a level of stochasticism incorporated into it. Seems that failures should occur to some level, but more investigation is required.

Here are the two runs that have hit this:
1. https://github.com/cilium/cilium/actions/runs/18934561156/job/54058216154
2. https://github.com/cilium/cilium/actions/runs/19071506098/job/54475532408

The only artifacts generated during the test were the junits zip:
[cilium-junits.zip](https://github.com/user-attachments/files/23346376/cilium-junits.zip)

--- a/bpf/tests/bpf_nat_tests.c
+++ b/bpf/tests/bpf_nat_tests.c
@@ -1421,7 +1421,7 @@ int test_nat4_port_allocation_tcp_check(struct __ctx_buff *ctx)
 	assert(cb_ctx.fail_thres >= SNAT_TEST_ITERATIONS * 7 / 10);
 
 	/* Only occasional failures at 50% of the test. */
-	assert(retries_50percent[SNAT_COLLISION_RETRIES] < 15);
+	assert(retries_50percent[SNAT_COLLISION_RETRIES] < 25);
 
 	/* Less than 7% of failures at 75% of the test. */
 	assert(retries_75percent[SNAT_COLLISION_RETRIES] < SNAT_TEST_ITERATIONS * 75 * 7 / 10000);
@@ -1548,7 +1548,7 @@ int test_nat4_port_allocation_udp_check(struct __ctx_buff *ctx)
 	assert(cb_ctx.fail_thres >= SNAT_TEST_ITERATIONS * 7 / 10);
 
 	/* Only occasional failures at 50% of the test. */
-	assert(retries_50percent[SNAT_COLLISION_RETRIES] < 15);
+	assert(retries_50percent[SNAT_COLLISION_RETRIES] < 25);
 
 	/* Less than 7% of failures at 75% of the test. */
 	assert(retries_75percent[SNAT_COLLISION_RETRIES] < SNAT_TEST_ITERATIONS * 75 * 7 / 10000);

